keyboards/kprepublic/bm40hsrgb/rev2/config.h

```diff
@@ -0,0 +1,38 @@
+/* Copyright 2020 tominabox1
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#ifdef RGB_DI_PIN
+#    define RGB_MATRIX_KEYPRESSES // reacts to keypresses
+#endif
+#ifdef RGB_MATRIX_ENABLE
+// IS31FL3733 configuration
+#    define DRIVER_COUNT 1
+#    define DRIVER_ADDR_1 0b1010000
+#    define DRIVER_1_LED_TOTAL 47
+#    define ISSI_LED_TOTAL DRIVER_1_LED_TOTAL
+// Underglow LEDs are WS2812, but someone might want to use RGBLIGHT for them;
+// don't use those LEDs in RGB Matrix in that case.
+#    ifdef RGBLIGHT_ENABLE
+#        define WS2812_LED_TOTAL 0
+#    else
+#        define WS2812_LED_TOTAL 6
+#    endif
+#    define DRIVER_LED_TOTAL (ISSI_LED_TOTAL + WS2812_LED_TOTAL)
+#    define RGB_MATRIX_LED_COUNT (DRIVER_LED_TOTAL)
+
+#    define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```

Comment:
```suggestion
#ifdef RGB_MATRIX_ENABLE
// IS31FL3733 configuration
#    define DRIVER_COUNT 1
#    define DRIVER_ADDR_1 0b1010000
// Underglow LEDs are WS2812, but someone might want to use RGBLIGHT for them;
// don't use those LEDs in RGB Matrix in that case.
#    ifdef RGBLIGHT_ENABLE
#        define WS2812_LED_TOTAL 0
#    else
#        define WS2812_LED_TOTAL 6
#    endif
#    define RGB_MATRIX_LED_COUNT 47 + WS2812_LED_TOTAL
#    define RGB_MATRIX_KEYPRESSES
#    define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```
Although, after looking through the custom code, I don't really see a need for `WS2812_LED_TOTAL` and the likely confusion it will cause?

Reply:
> I don't really see a need for WS2812_LED_TOTAL and the likely confusion it will cause?

The intention of this custom driver code is to allow either of: using the rgb_matrix feature for the per-key RGB LEDs (issi), while also using rgblight for the underglow; *or* to use rgb_matrix for all the RGB LEDs.

Hence, `WS2812_LED_TOTAL > 0` is "the rgb_matrix driver controls the 6 underglow RGB LEDs"; `WS2812_LED_TOTAL` is `0` for "rgblight controls those LEDs".

https://github.com/qmk/qmk_firmware/pull/16689#issuecomment-1513505823

(I can't take credit for this clever code; some of the other KPRepublic PCBs follow a similar format).